File: Array/594-Longest-Harmonious-Subsequence.java
import java.util.HashMap;

class Solution {
    public int findLHS(int[] nums) {
        HashMap<Integer, Integer> count = new HashMap<>();
        int maxSubstring = 0;

        for (int i = 0; i < nums.length; i++) {
            count.put(nums[i], count.getOrDefault(nums[i], 0) + 1);
        }

        for (int key : count.keySet()) {
            if (count.containsKey(key + 1)) {
                int length = count.get(key) + count.get(key + 1);
                maxSubstring = Math.max(maxSubstring, length);
            }
        }

        return maxSubstring;
    }
}

File: Array/594-Longest-Harmonious-Subsequence.ts
function findLHS(nums: number[]): number {
    let subLength = 0;
    let count: Map<number, number> = new Map();

    for (let i = 0; i < nums.length; i++) {
        count.set(nums[i], (count.get(nums[i]) || 0) + 1);
    }

    for (let key of count.keys()) {
        if (count.has(key + 1)) {
            const length = (count.get(key) || 0) + (count.get(key + 1) || 0);
            subLength = Math.max(subLength, length);
        }
    }

    return subLength;
};